var/spack/repos/builtin/packages/py-coverage/package.py

```diff
@@ -29,8 +29,9 @@ class PyCoverage(PythonPackage):
     """ Testing coverage checker for python """
 
     homepage = "http://nedbatchelder.com/code/coverage/"
-    url      = "https://pypi.python.org/packages/source/c/coverage/coverage-4.0a6.tar.gz"
+    url      = "https://github.com/nedbat/coveragepy/archive/coverage-4.3.4.tar.gz"
 
+    version('4.3.4', 'd347766b06bbb4fd0bc822014b7cfb0a')
     version('4.0a6', '1bb4058062646148965bef0796b61efc')
 
```

Comment:
It is important that URLs NOT be changed without good cause.  As @adamjstewart mentioned, we normally download Python stuff from PyPI; this recommendation is also supported by the Py-Coverage docs (pointed to from the PyPI page).  Nowhere can I verify, starting from PyPI, that `github.com/netdbat` belongs to the author of Py-Coverage.  Moreover, the checksum reported for the `4.3.4` tarball from GitHub is different from that reported on the PyPI page for the same tarball.  **Conclusion**: For all these reasons, please revert `py-coverage` back to using PyPI.  Something like this:

```
    homepage = "http://nedbatchelder.com/code/coverage/"
 
    version('4.3.4', '89759813309185efcf4af8b9f7762630',
        url='https://pypi.python.org/packages/6e/33/01cb50da2d0582c077299651038371dba988248058e03c7a7c4be0c84c40/coverage-4.3.4.tar.gz')
     version('4.0a6', '1bb4058062646148965bef0796b61efc',
        url='https://pypi.python.org/packages/source/c/coverage/coverage-4.0a6.tar.gz')
```
Someone should also verify that the URL for `4.0a6` is still valid.  If not ,feel free to just drop the version rather than discovering the new URL at PyPI.